src/http_tracker_connection.cpp

```diff
@@ -168,10 +168,10 @@ namespace libtorrent
 #if TORRENT_USE_I2P
 			if (i2p && tracker_req().i2pconn)
 			{
-				url += "&ip=";
-				url += escape_string(tracker_req().i2pconn->local_endpoint().c_str()
-					, tracker_req().i2pconn->local_endpoint().size());
-				url += ".i2p";
+				if (tracker_req().i2pconn->local_endpoint().empty())
+					fail(error_code(errors::no_i2p_endpoint), -1, "Waiting for i2p acceptor from SAM bridge", 5);
```

Comment:
Indeed... when I tried to return (before failing), I had some issues with the timer as it didn't start and just nothing happened. I'll add this.